packages/calypso-e2e/src/lib/components/domain-search-component.ts

```diff
@@ -81,16 +82,17 @@ export class DomainSearchComponent {
 
 		await target.click();
 
-		await this.page.getByRole( 'button' ).filter( { hasText: 'Continue' } ).waitFor();
+		try {
+			await this.page.getByRole( 'button' ).filter( { hasText: 'Continue' } ).waitFor();
+		} catch ( exception ) {
+			// Don't do anything since the button might never appear in some cases (e.g. with-theme flow)
+		}
```

Comment:
Can we use anything on the page (or the current URL) to determine the current case? I.e. whether it's the `with-theme` flow or something else? Wrapping the locator with try/catch is generally bad practice as we rely solely on a timeout value which can produce flaky results.

Reply:
Ah, I can see that we use Promise.race for the `Continue` button click vs the target URL. In that case, this entire try/catch block can be removed! 😄 
```suggestion
```

Reply:
Thanks for the context. From the functionality perspective, this part doesn't make sense since it doesn't address any delays in the UI - it's already handled by the button `click` util below. If anything, we might need to extend the timeout on that click, but nothing else should be necessary here.